packages/build/src/plugins/child/mutations.js

```diff
@@ -23,13 +23,21 @@ const preventObjectMutations = function (value, keys) {
   }
 
   if (isPlainObj(value)) {
-    const object = mapObj(value, (key, item) => [key, preventObjectMutations(item, [...keys, key])])
+    const object = mapObj(value, (key, item) => [key, preventObjectMutations(item, [...keys, getChildKey(key, keys)])])
     return addProxy(object, keys)
   }
 
   return value
 }
 
+// Some properties are user-defined, i.e. we need to replace them with a "*" token
```

Comment:
```suggestion
// Some properties can take any value defined by the user, as opposed to belonging
// to a static list of names. In order to compare them against the list of mutable
// keys, we need to normalize them by replacing the dynamic portion with "*".
```

The description of this function wasn't immediately clear to me. How do you feel about adding this change to the comment?

I also feel that there's a more descriptive name than `getChildKey` but I'm struggling to find one. 🤔

Reply:
Fixed :+1: